errext/errorset_test.go

```diff
@@ -0,0 +1,241 @@
+	errs1.Add(errors.New("error 2"))
+
+	var errs2 ErrorSet
+	errs2.Add(errors.New("error 3"))
+	errs2.Add(errors.New("error 4"))
+
+	// Append errs2 to errs1
+	errs1.Append(errs2)
+	assert.Equal(t, len(errs1), 4)
+	assert.Equal(t, errs1[0].Error(), "error 1")
+	assert.Equal(t, errs1[1].Error(), "error 2")
+	assert.Equal(t, errs1[2].Error(), "error 3")
+	assert.Equal(t, errs1[3].Error(), "error 4")
+
+	// Append empty ErrorSet
+	var errs3 ErrorSet
+	errs1.Append(errs3)
+	assert.Equal(t, len(errs1), 4)
+}
+
+func TestErrorSetIsEmpty(t *testing.T) {
+	var errs ErrorSet
+
+	// Empty ErrorSet
+	assert.Equal(t, errs.IsEmpty(), true)
+
+	// Add an error
+	errs.Add(errors.New("error"))
+	assert.Equal(t, errs.IsEmpty(), false)
+
+	// Create new ErrorSet with errors
+	errs2 := ErrorSet{errors.New("error 1"), errors.New("error 2")}
+	assert.Equal(t, errs2.IsEmpty(), false)
+}
+
+func TestErrorSetJoin(t *testing.T) {
+	var errs ErrorSet
+
+	// Empty ErrorSet
+	result := errs.Join(", ")
+	assert.Equal(t, result, "")
+
+	// Single error
+	errs.Add(errors.New("error 1"))
+	result = errs.Join(", ")
+	assert.Equal(t, result, "error 1")
+
+	// Multiple errors with comma separator
+	errs.Add(errors.New("error 2"))
+	errs.Add(errors.New("error 3"))
+	result = errs.Join(", ")
+	assert.Equal(t, result, "error 1, error 2, error 3")
+
+	// Multiple errors with different separator
+	result = errs.Join(" | ")
+	assert.Equal(t, result, "error 1 | error 2 | error 3")
+
+	// Multiple errors with newline separator
+	result = errs.Join("\n")
+	assert.Equal(t, result, "error 1\nerror 2\nerror 3")
+}
+
+func TestErrorSetJoinedError(t *testing.T) {
+	var errs ErrorSet
+
+	// Empty ErrorSet
+	joined := errs.JoinedError(", ")
+	assert.Equal(t, joined.Error(), "")
+
+	// Single error
+	errs.Add(errors.New("error 1"))
+	joined = errs.JoinedError(", ")
+	assert.Equal(t, joined.Error(), "error 1")
+
+	// Multiple errors
+	errs.Add(errors.New("error 2"))
+	errs.Add(errors.New("error 3"))
+	joined = errs.JoinedError("; ")
+	assert.Equal(t, joined.Error(), "error 1; error 2; error 3")
+}
+
+func TestJoinedErrorError(t *testing.T) {
+	// Test the Error() method of joinedError directly
+	je := joinedError{
+		errs:      []error{errors.New("error 1"), errors.New("error 2")},
+		separator: " - ",
+	}
+	assert.Equal(t, je.Error(), "error 1 - error 2")
+
+	// Empty error list
+	je2 := joinedError{
+		errs:      []error{},
+		separator: ", ",
+	}
+	assert.Equal(t, je2.Error(), "")
+}
+
+func TestJoinedErrorUnwrap(t *testing.T) {
+	err1 := errors.New("error 1")
+	err2 := errors.New("error 2")
+	err3 := errors.New("error 3")
+
+	var errs ErrorSet
+	errs.Add(err1)
+	errs.Add(err2)
+	errs.Add(err3)
+
+	joined := errs.JoinedError(", ")
+
+	// Unwrap should return the original errors
+	unwrapped := joined.(interface{ Unwrap() []error }).Unwrap()
+	assert.Equal(t, len(unwrapped), 3)
+	assert.Equal(t, unwrapped[0], err1)
+	assert.Equal(t, unwrapped[1], err2)
+	assert.Equal(t, unwrapped[2], err3)
```

Comment:
The fact that we implement with package errors through `Unwrap() []error` is an implementation detail and should not be explicitly checked in the test suite. The Unwrap implementation gets test coverage through looking at errors.Is() further below in this test, and that is enough. Same for the other test below that looks at Unwrap: It should look at a high-level interface like errors.Is() or errors.As().